guest/x86/flat.lds

```diff
@@ -1,8 +1,9 @@
 SECTIONS
 {
-    . = 4M + SIZEOF_HEADERS;
+    . = 4M;
```

Comment:
Can we make the entry to 4M + SIZEOF_HEADERS + 0xC?

Reply:
Other changes looks good to me.

Reply:
SIZEOF_HEADERS may change from time to time (e.g. due to the number of sections in the linked executable). This makes the identification of the entry address impossible as we use an objdump'ed raw binary for partition mode.

How about we have two lds, one for single-test executable that should to be loaded as an ELF and the other for the stitched executable that have a fixed entry address and can be loaded as a raw binary?

Reply:
Two lds is complicated. Can we have a header added at the beginning of each raw binary which include the binary entry in fixed offset?

Reply:
Sorry for the late response.

I have restored the original placement of sections. We can rely on the multiboot header which is always 12-byte and right ahead of the entry point.

Also the .comment and .note sections are discarded as it has been met that ignoring these sections in the linker script leads to a very small bss_start (mostly <1K), which leads the test program to crash when it tries to initialize its .bss section.